src/main/java/org/opensearch/knn/index/query/filtered/NestedFilteredIdsKNNIterator.java

```diff
@@ -38,20 +38,24 @@ public NestedFilteredIdsKNNIterator(
      */
     @Override
     public int nextDoc() throws IOException {
-        if (currentPos >= filterIdsArray.length) {
+        if (docId == DocIdSetIterator.NO_MORE_DOCS) {
             return DocIdSetIterator.NO_MORE_DOCS;
         }
+
         currentScore = Float.NEGATIVE_INFINITY;
-        int currentParent = parentBitSet.nextSetBit(filterIdsArray[currentPos]);
+        int currentParent = parentBitSet.nextSetBit(docId);
         int bestChild = -1;
-        while (currentPos < filterIdsArray.length && filterIdsArray[currentPos] < currentParent) {
-            binaryDocValues.advance(filterIdsArray[currentPos]);
+
+        while (docId != DocIdSetIterator.NO_MORE_DOCS && docId < currentParent) {
+            System.out.println("Before DocID" + docId + " currentParent:" + currentParent);
```

Comment:
remove this line too.

Reply:
FIXED at https://github.com/opensearch-project/k-NN/pull/1402/commits/a48a928eba4b468bf64f6adc1f82f88e38749dbc